test/parallel/test-timers-api-refs.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+const common = require('../common');
+const timers = require('timers');
+
+// delete global APIs to make sure they're not relied on by the internal timers
+// code
+delete global.setTimeout;
+delete global.clearTimeout;
+delete global.setInterval;
+delete global.clearInterval;
+delete global.setImmediate;
+delete global.clearImmediate;
```

Comment:
You've also changed `active`, but I'm not comfortable enforcing that in the tests. We should have a policy if module exports are always overridable.